cognite/client/utils/_time.py

```diff
@@ -164,44 +165,54 @@ def timestamp_to_ms(timestamp: int | float | str | datetime) -> int:
 
 
 TIME_ATTRIBUTES = {
-    "start_time",
+    "created_time",
+    "creation_time",
+    "deleted_time",
     "end_time",
+    "expiration_time",
+    "last_failure",
+    "last_indexed_time",
+    "last_seen",
+    "last_success",
     "last_updated_time",
-    "created_time",
-    "timestamp",
+    "modified_time",
     "scheduled_execution_time",
     "source_created_time",
     "source_modified_time",
+    "start_time",
+    "timestamp",
+    "uploaded_time",
 }
+TIME_ATTRIBUTES |= set(map(to_camel_case, TIME_ATTRIBUTES))
 
 
-def _convert_time_attributes_in_dict(item: dict) -> dict:
-    new_item = {}
-    for k, v in item.items():
-        if k in TIME_ATTRIBUTES:
-            try:
-                v = str(ms_to_datetime(v).replace(tzinfo=None))
-            except ValueError:
-                pass
-        new_item[k] = v
-    return new_item
+def _convert_time_attrs_in_dict(item: dict, to_isoformat: bool) -> dict:
+    for k in TIME_ATTRIBUTES.intersection(item):
+        try:
+            if to_isoformat:
+                item[k] = ms_to_datetime(item[k]).isoformat(sep=" ", timespec="milliseconds")
+            else:
+                item[k] = ms_to_datetime(item[k])
+        except ValueError:
+            pass
```

Comment:
```suggestion
        with suppress(ValueError):
            if to_isoformat:
                item[k] = ms_to_datetime(item[k]).isoformat(sep=" ", timespec="milliseconds")
            else:
                item[k] = ms_to_datetime(item[k])
```

And remember to `from contextlib import suppress`

Reply:
I went with try-except as `suppress` has quite the larger overhead (with enter and exit methods) - and we call this in a loop. *But I prefer suppress* 😎